code/datums/components/cult_ritual_item.dm

```diff
@@ -0,0 +1,474 @@
+/*
+ * Component for items that are used by cultists to conduct rituals.
+ *
+ * - Draws runes, including the rune to summon Nar'sie.
+ * - Purges cultists of holy water on attack.
+ * - (Un/re)anchors cult structures when hit.
+ * - Instantly destroys cult girders on hit.
+ */
+/datum/component/cult_ritual_item
+	/// Whether we are currently being used to draw a rune.
+	var/drawing_a_rune = FALSE
+	/// The message displayed on examine, if supplied.
+	var/examine_message
+	/// A list of turfs that we scribe runes at double speed on
+	var/list/turfs_that_boost_us
+	/// A list of all shields surrounding us while drawing runes.
+	var/list/obj/structure/emergency_shield/cult/narsie/shields
+	/// The action associated with our item
+	var/datum/action/item_action/linked_action
+
+/datum/component/cult_ritual_item/Initialize(
+	examine_message,
+	action = /datum/action/item_action/cult_dagger,
+	turfs_that_boost_us = /turf/open/floor/engine/cult,
+	)
+
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	// Initialize all our rune types if it wasn't already done.
+	if(!LAZYLEN(GLOB.rune_types))
+		for(var/obj/effect/rune/rune as anything in subtypesof(/obj/effect/rune))
+			if(!initial(rune.can_be_scribed))
+				continue
+			LAZYSET(GLOB.rune_types, initial(rune.cultist_name), rune) // Uses the cultist name for displaying purposes
```

Comment:
this should really be done when the rune_types list is initialized

Reply:
In the past it didn't initialize rune types unless a ritual dagger was initialized, presumably to save memory not filling a list that wouldn't be used? I can't change that though.

Reply:
why not? you can run procs when the list is initialized

Reply:
My bad, I meant to say "I can change that". Will do.